Sources/Testing/Events/Event.swift

```diff
@@ -143,37 +143,75 @@ extension Event {
   ///
   /// - Parameters:
   ///   - event: An event that needs to be handled.
-  public typealias Handler = @Sendable (Event) -> Void
+  ///   - context: The context associated with the event.
+  public typealias Handler = @Sendable (_ event: consuming Event, _ context: consuming Context) -> Void
+
+  /// A type which provides context about a posted ``Event``.
+  ///
+  /// An instance of this type is provided along with each ``Event`` that is
+  /// passed to an ``Event/Handler``.
+  public struct Context: Sendable {
+    /// The runner which ran the test which caused this instance's associated
+    /// event to be posted.
+    ///
+    /// This runner may be used within a ``Event/Handler`` to retrieve the
+    /// ``Test`` (if any) that a particular ``Event`` references via its
+    /// ``Event/testID`` property.
+    @_spi(ExperimentalTestRunning)
+    public var runner: Runner
```

Comment:
Is this property's value correct when we temporarily pop the current context for self-hosted tests? (We need a unit test for that.)

Reply:
Addressed that [here](https://github.com/apple/swift-testing/pull/39#discussion_r1342940290)!